src/pkcs11/openssl.c

```diff
@@ -272,6 +277,7 @@ sc_pkcs11_register_openssl_mechanisms(struct sc_pkcs11_card *p11card)
 		CRYPTO_set_locking_callback(locking_cb);
 #endif
 #endif /* !defined(OPENSSL_NO_ENGINE) */
+#endif /* OPENSSL_VERSION_NUMBER >= 0x30000000L && !defined(LIBRESSL_VERSION_NUMBER) */
```

Comment:
The part `!defined(LIBRESSL_VERSION_NUMBER)` should not go here, but on the line 268 with the `!OPENSSL_NO_STATIC_ENGINE && !OPENSSL_NO_GOST ` condition to match the opening.

Reply:
\#endif changed to match starting \#if  in https://github.com/OpenSC/OpenSC/pull/2506/commits/723f559db0bf6d4e7731afe2b77e973662867b7c